commands/box/wallet/root.go

```diff
@@ -179,3 +209,72 @@ func dumpPrivKeyCmdFunc(cmd *cobra.Command, args []string) {
 	}
 	fmt.Printf("Address: %s\nPrivate Key: %s\n", addr, privateKey)
 }
+func dumpwallet(cmd *cobra.Command, args []string) {
+	fmt.Println("dumpwallet called")
+	if len(args) < 1 {
+		fmt.Println("file name needed")
+		return
+	}
+	if len(args[0]) <= 4 {
+		fmt.Println("file name error")
+		return
+	}
+	if args[0][len(args[0])-4:] != ".txt" {
+		fmt.Println("Input file type must be in txt file format")
+		return
+	}
+	_, err := os.Stat(args[0])
+	if err == nil {
+		fmt.Println("The file already exists and the command will overwrite the contents of the file.")
+	}
+	file, err := os.OpenFile(args[0], os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0666)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	file.Write([]byte("# Wallet dump created by ContentBox\n" + "# * Created on " + time.Now().Format("2006-01-02 15:04:05") + "\n"))
+
+	addrs := make([]string, 0)
+	wltMgr, err := wallet.NewWalletManager(walletDir)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	for _, acc := range wltMgr.ListAccounts() {
+		addrs = append(addrs, acc.Addr())
+	}
+	success_num := 0
+	failed_num := 0
+	for _, x := range addrs {
+		addr := x
+		wltMgr, err := wallet.NewWalletManager(walletDir)
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		fmt.Print("addr: " + addr)
```

Comment:
output format!!!

Reply:
Behind another loop, so you need to type it in one line.